utils/node_local_test.py

```diff
@@ -0,0 +1,798 @@
+    return ret
+
+def symlink_file(a, b):
+    """Create a symlink from a to b"""
+    if os.path.exists(a):
+        os.remove(a)
+    os.symlink(b, a)
+
+class ValgrindHelper():
+
+    """Class for running valgrind commands
+
+    This helps setup the command line required, and
+    performs log modification after the fact to assist
+    Jenkins in locating the source code.
+    """
+    instance_num = 0
+
+    def __init__(self, logid=None):
+
+        if not logid:
+            self.__class__.instance_num += 1
+            logid = self.__class__.instance_num
+
+        self.xml_file = 'dnt.{}.memcheck'.format(logid)
+
+        self.src_dir = '{}/'.format(os.path.realpath(
+            os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))
+
+    def get_cmd_prefix(self):
+        """Return the command line prefix"""
+        cmd = ['valgrind', '--quiet']
+
+        cmd.extend(['--leak-check=full', '--show-leak-kinds=all'])
+
+        s_arg = '--suppressions='
+        cmd.extend(['{}{}'.format(s_arg,
+                                  os.path.join('src',
+                                               'cart',
+                                               'utils',
+                                               'memcheck-cart.supp')),
+                    '{}{}'.format(s_arg,
+                                  os.path.join('utils',
+                                               'memcheck-daos-client.supp'))])
+
+        cmd.extend(['--xml=yes',
+                    '--xml-file={}'.format(self.xml_file)])
+        return cmd
+
+    def convert_xml(self):
+        """Modify the xml file"""
+        fd = open(self.xml_file, 'r')
+        ofd = open('{}.xml'.format(self.xml_file), 'w')
+        for line in fd:
+            if self.src_dir in line:
+                ofd.write(line.replace(self.src_dir, ''))
+            else:
+                ofd.write(line)
+
+class DFuse():
+    """Manage a dfuse instance"""
+
+    instance_num = 0
+
+    def __init__(self, daos, conf, pool=None, container=None):
+        self.dir = '/tmp/dfs_test'
+        self.pool = pool
+        self.valgrind_file = None
+        self.container = container
+        self.conf = conf
+        self._daos = daos
+        self._sp = None
+
+        log_file = tempfile.NamedTemporaryFile(prefix='dnt_dfuse_',
+                                               suffix='.log', delete=False)
+        self.log_file = log_file.name
+
+        symlink_file('/tmp/dfuse_latest.log', self.log_file)
+
+        self.valgrind = None
+
+        if not os.path.exists(self.dir):
+            os.mkdir(self.dir)
+
+    def start(self, v_hint=None):
+        """Start a dfuse instance"""
+        dfuse_bin = os.path.join(self.conf['PREFIX'], 'bin', 'dfuse')
+        my_env = os.environ.copy()
+
+        single_threaded = False
+        caching = False
+
+        pre_inode = os.stat(self.dir).st_ino
+
+        my_env['CRT_PHY_ADDR_STR'] = 'ofi+sockets'
+        my_env['OFI_INTERFACE'] = 'eth0'
```

Comment:
Optional: will be better if we use ib0. And this applies to whole of testing if possible.

Reply:
This runs on a single VM which is why it's so fast, ib0 will not exist.  I was using "lo" for a bit but the setup depends on what the VM has available.  My suspicion is as long as it's a valid interface it doesn't matter.